components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -209,6 +211,19 @@ static esp_err_t esp_websocket_client_abort_connection(esp_websocket_client_hand
         client->reconnect_tick_ms = _tick_get_ms();
         ESP_LOGI(TAG, "Reconnect after %d ms", client->wait_timeout_ms);
     }
+
+    if (error_type == WEBSOCKET_ERROR_TYPE_TCP_TRANSPORT)
+    {
+        client->error_handle.esp_tls_last_esp_err =
+            esp_tls_get_and_clear_last_error(esp_transport_get_error_handle(client->transport),
+                                             &client->error_handle.esp_tls_stack_err,
+                                             &client->error_handle.esp_tls_cert_verify_flags);
+        client->error_handle.esp_transport_sock_errno = esp_transport_get_errno(client->transport);
```

Comment:
Here we copy the errors from `tcp_transport` to client's struct. Could we possibly move this code to the dispatch function? Maybe we could copy it directly to the event struct and avoid unnecessary copy?